File: apps/web/src/types/review.ts
/**
 * ReviewSystemTypeDefinition
 */

// ReviewStatus
export type ReviewStatus = 
 | "pending" // Pending Review
 | "in_review" // Review
 | "approved" // alreadyVia
 | "rejected" // alreadyDeny
 | "revision" // needEdit
 | "cancelled"; // Cancelled

// ReviewitemType
export type ReviewItemType = 
 | "agent" // Agent Review
 | "workflow" // WorkflowReview
 | "template" // TemplateReview
 | "user" // UserReview
 | "content"; // ContentReview

// ReviewPriority
export type ReviewPriority = 
 | "low" // Priority
 | "normal" // NormalPriority
 | "high" // Priority
 | "urgent"; // Urgent

// Review
export interface Reviewer {
 id: string;
 userId: string;
 role: string;
 displayName?: string;
 allowedTypes: ReviewItemType[];
 maxDailyReviews: number;
 currentWorkload: number;
 isActive: boolean;
 totalReviews: number;
 approvedCount: number;
 rejectedCount: number;
 avgReviewTime: number; // AverageReviewTime(s)
 createdAt: string;
 updatedAt: string;
 user?: {
 id: string;
 username: string;
 email: string;
 avatar?: string;
 };
}

// ReviewQueue
export interface ReviewQueueItem {
 id: string;
 itemType: ReviewItemType;
 itemId: string;
 submitterId: string;
 reviewerId?: string;
 assignedAt?: string;
 status: ReviewStatus;
 priority: ReviewPriority;
 title: string;
 description?: string;
 snapshot: Record<string, unknown>;
 submissionNote?: string;
 resultNote?: string;
 resultData?: Record<string, unknown>;
 revisionCount: number;
 revisionNote?: string;
 version: number;
 submittedAt: string;
 reviewedAt?: string;
 createdAt: string;
 updatedAt: string;
 submitter?: {
 id: string;
 username: string;
 email: string;
 avatar?: string;
 };
 reviewer?: Reviewer;
}

// ReviewRecord
export interface ReviewRecord {
 id: string;
 queueId: string;
 reviewerId: string;
 action: string; // assign, review, approve, reject, request_revision
 fromStatus?: ReviewStatus;
 toStatus: ReviewStatus;
 comment?: string;
 details?: Record<string, unknown>;
 durationMs?: number;
 createdAt: string;
 reviewer?: Reviewer;
}

// ReviewComment
export interface ReviewComment {
 id: string;
 queueId: string;
 userId: string;
 content: string;
 commentType: string; // comment, question, suggestion, issue
 targetPath?: string;
 parentId?: string;
 isResolved: boolean;
 resolvedAt?: string;
 resolvedBy?: string;
 createdAt: string;
 updatedAt: string;
 user?: {
 id: string;
 username: string;
 email: string;
 avatar?: string;
 };
 replies?: ReviewComment[];
}

// ReviewCheck
export interface ChecklistItem {
 id: string;
 label: string;
 required: boolean;
 category: string;
 checked?: boolean;
}

// ReviewCheckTemplate
export interface ReviewChecklist {
 id: string;
 name: string;
 description?: string;
 itemType: ReviewItemType;
 items: ChecklistItem[];
 isActive: boolean;
 isDefault: boolean;
 createdAt: string;
 updatedAt: string;
}

// ReviewStatistics
export interface ReviewStats {
 pending: number;
 inReview: number;
 approved: number;
 rejected: number;
 revision: number;
 total: number;
 avgReviewTimeHours: number;
}

// =====================
// RequestParameterType
// =====================

// ReviewQueueListRequestParameter
export interface ListReviewQueueParams {
 itemType?: ReviewItemType;
 status?: ReviewStatus;
 priority?: ReviewPriority;
 submitterId?: string;
 reviewerId?: string;
 page?: number;
 pageSize?: number;
 startDate?: string;
 endDate?: string;
}

// SubmitReviewRequest
export interface SubmitReviewRequest {
 itemType: ReviewItemType;
 itemId: string;
 title: string;
 description?: string;
 submissionNote?: string;
 priority?: ReviewPriority;
}

// ReviewActionRequest
export interface ReviewActionRequest {
 action: "approve" | "reject" | "request_revision";
 comment?: string;
 details?: Record<string, unknown>;
 checklist?: ChecklistItem[];
}

// AllocateReviewRequest
export interface AssignReviewerRequest {
 reviewerId: string;
}

// CreateCommentRequest
export interface CreateCommentRequest {
 content: string;
 commentType?: string;
 targetPath?: string;
 parentId?: string;
}

// =====================
// ResponseType
// =====================

// ReviewQueueListResponse
export interface ListReviewQueueResponse {
 success: boolean;
 data: ReviewQueueItem[];
 meta: {
 total: number;
 page: number;
 pageSize: number;
 };
}

// ReviewDetailsResponse
export interface GetReviewDetailResponse {
 success: boolean;
 data: ReviewQueueItem & {
 records: ReviewRecord[];
 comments: ReviewComment[];
 };
}

// ReviewStatisticsResponse
export interface GetReviewStatsResponse {
 success: boolean;
 data: ReviewStats;
}

// SubmitReviewResponse
export interface SubmitReviewResponse {
 success: boolean;
 data: ReviewQueueItem;
 message: string;
}

// ReviewActionResponse
export interface ReviewActionResponse {
 success: boolean;
 data: ReviewQueueItem;
 message: string;
}

// ReviewCheckListResponse
export interface GetChecklistResponse {
 success: boolean;
 data: ReviewChecklist;
}

// CommentActionResponse
export interface CommentResponse {
 success: boolean;
 data: ReviewComment;
 message: string;
}

// ReviewListResponse
export interface ListReviewersResponse {
 success: boolean;
 data: Reviewer[];
 meta: {
 total: number;
 page: number;
 pageSize: number;
 };
}

File: apps/server/internal/domain/entity/review.go
package entity

// review.go — Minimal stubs for review types used by DB schema migration approval.
// Full review domain has been removed; only types referenced by core services remain.

import (
	"time"

	"github.com/google/uuid"
)

// ReviewItemType 审核项类型
type ReviewItemType string

const (
	ReviewItemTypeDBSchema ReviewItemType = "db_schema"
)

// ReviewStatus 审核状态
type ReviewStatus string

const (
	ReviewStatusPending  ReviewStatus = "pending"
	ReviewStatusInReview ReviewStatus = "in_review"
	ReviewStatusApproved ReviewStatus = "approved"
	ReviewStatusRejected ReviewStatus = "rejected"
	ReviewStatusRevision ReviewStatus = "revision"
)

// ReviewPriority 审核优先级
type ReviewPriority string

const (
	ReviewPriorityNormal ReviewPriority = "normal"
)

// ReviewQueue 审核队列
type ReviewQueue struct {
	ID             uuid.UUID      `gorm:"type:char(36);primaryKey" json:"id"`
	ItemType       ReviewItemType `gorm:"type:varchar(50);not null;index" json:"item_type"`
	ItemID         uuid.UUID      `gorm:"type:char(36);not null;index" json:"item_id"`
	SubmitterID    uuid.UUID      `gorm:"type:char(36);not null;index" json:"submitter_id"`
	ReviewerID     *uuid.UUID     `gorm:"type:char(36);index" json:"reviewer_id,omitempty"`
	Status         ReviewStatus   `gorm:"type:varchar(20);not null;default:pending;index" json:"status"`
	Priority       ReviewPriority `gorm:"type:varchar(20);not null;default:normal" json:"priority"`
	Title          string         `gorm:"type:varchar(200)" json:"title"`
	Description    *string        `gorm:"type:text" json:"description,omitempty"`
	SubmissionNote *string        `gorm:"type:text" json:"submission_note,omitempty"`
	Snapshot       JSON           `gorm:"type:json" json:"snapshot,omitempty"`
	CreatedAt      time.Time      `json:"created_at"`
	UpdatedAt      time.Time      `json:"updated_at"`
}

// ReviewRecord 审核记录
type ReviewRecord struct {
	ID         uuid.UUID     `gorm:"type:char(36);primaryKey" json:"id"`
	QueueID    uuid.UUID     `gorm:"type:char(36);not null;index" json:"queue_id"`
	ReviewerID uuid.UUID     `gorm:"type:char(36);not null" json:"reviewer_id"`
	Action     string        `gorm:"type:varchar(50);not null" json:"action"`
	FromStatus *ReviewStatus `gorm:"type:varchar(20)" json:"from_status,omitempty"`
	ToStatus   ReviewStatus  `gorm:"type:varchar(20);not null" json:"to_status"`
	Comment    *string       `gorm:"type:text" json:"comment,omitempty"`
	CreatedAt  time.Time     `json:"created_at"`
}

// ReviewComment 审核评论
type ReviewComment struct {
	ID         uuid.UUID  `gorm:"type:char(36);primaryKey" json:"id"`
	QueueID    uuid.UUID  `gorm:"type:char(36);not null;index" json:"queue_id"`
	UserID     uuid.UUID  `gorm:"type:char(36);not null" json:"user_id"`
	Content    string     `gorm:"type:text;not null" json:"content"`
	ResolvedAt *time.Time `json:"resolved_at,omitempty"`
	ResolvedBy *uuid.UUID `gorm:"type:char(36)" json:"resolved_by,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
}

// Reviewer 审核员
type Reviewer struct {
	ID            uuid.UUID `gorm:"type:char(36);primaryKey" json:"id"`
	UserID        uuid.UUID `gorm:"type:char(36);not null;uniqueIndex" json:"user_id"`
	Active        bool      `gorm:"default:true" json:"active"`
	TotalReviews  int       `gorm:"default:0" json:"total_reviews"`
	ApprovedCount int       `gorm:"default:0" json:"approved_count"`
	RejectedCount int       `gorm:"default:0" json:"rejected_count"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// ReviewStats 审核统计
type ReviewStats struct {
	ItemType ReviewItemType `json:"item_type"`
	Status   ReviewStatus   `json:"status"`
	Count    int64          `json:"count"`
}

// ReviewChecklist 审核检查项模板
type ReviewChecklist struct {
	ID        uuid.UUID      `gorm:"type:char(36);primaryKey" json:"id"`
	ItemType  ReviewItemType `gorm:"type:varchar(50);not null;uniqueIndex" json:"item_type"`
	Items     JSON           `gorm:"type:json" json:"items"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
}
